crates/symbolicator/src/services/download/mod.rs

```diff
@@ -289,12 +283,11 @@ async fn download_stream(
         Ok(DownloadStatus::Completed)
     };
 
-    match timeout {
-        Some(timeout) => tokio::time::timeout(timeout, future)
-            .await
-            .map_err(|_| DownloadError::Canceled)?,
-        None => future.await,
-    }
+    // we fall back to a hard timeout of 1h
+    let timeout = timeout.unwrap_or(Duration::from_secs(60 * 60));
```

Comment:
Does clippy let you do this? This looks like it should be `unwrap_or_else`.

Reply:
I reverted this specific change, as we do already have a "global" download timeout which includes this future as well, so need for this ;-)